client/src/rpc_client.rs

```diff
@@ -1180,17 +1019,20 @@ impl RpcClient {
     }
 
     pub fn validator_exit(&self) -> ClientResult<bool> {
-        let response = self
-            .client
-            .send(&RpcRequest::ValidatorExit, Value::Null, 0)
-            .map_err(|err| err.into_with_command("ValidatorExit"))?;
-        serde_json::from_value(response)
-            .map_err(|err| ClientError::new_with_command(err.into(), "ValidatorExit"))
+        self.send(RpcRequest::ValidatorExit, Value::Null, 0)
     }
 
-    pub fn send(&self, request: &RpcRequest, params: Value, retries: usize) -> ClientResult<Value> {
+    pub fn send<T>(&self, request: RpcRequest, params: Value, retries: usize) -> ClientResult<T>
```

Comment:
This function signature change is responsible for most of the churn and de-duping in this PR
```diff
-    pub fn send(&self, request: &RpcRequest, params: Value, retries: usize) -> ClientResult<Value> {
+    pub fn send<T>(&self, request: RpcRequest, params: Value, retries: usize) -> ClientResult<T>
```